great_expectations/core/http.py

```diff
@@ -0,0 +1,60 @@
+import requests
+from requests.adapters import HTTPAdapter, Retry
+
+from great_expectations import __version__
+
+DEFAULT_TIMEOUT = 20
+
+
+class TimeoutHTTPAdapter(HTTPAdapter):
+    # https://stackoverflow.com/a/62044100
+    # Session-wide timeouts are not supported by requests
+    # but are discussed in detail here: https://github.com/psf/requests/issues/3070
+    def __init__(self, *args, **kwargs):
+        if "timeout" in kwargs:
+            self.timeout = kwargs["timeout"]
+            del kwargs["timeout"]
+        super().__init__(*args, **kwargs)
+
+    def send(self, request, **kwargs):
+        timeout = kwargs.get("timeout")
+        if timeout is None:
+            kwargs["timeout"] = self.timeout
+        return super().send(request, **kwargs)
+
+
+def session_factory(
```

Comment:
nit: I prefer `get_session` or `create_session` since the result isn't a factory but a session object.